src/integrationtests/java/com/aws/greengrass/integrationtests/ShadowManagerTest.java

```diff
@@ -247,4 +264,51 @@ void GIVEN_shadow_manager_WHEN_strategy_config_resets_THEN_respond_to_config_upd
         kernel.getContext().waitForPublishQueueToClear();
         assertThat(syncHandler.getOverallSyncStrategy(), instanceOf(PeriodicSyncStrategy.class));
     }
+
+    @Test
+    void GIVEN_cryptoKeyProviderService_WHEN_get_client_THEN_wait_for_cryptoservice(ExtensionContext context) throws InterruptedException, TLSAuthException, IoTDataPlaneClientCreationException {
+        ignoreExceptionOfType(context, TLSAuthException.class);
+        ignoreExceptionOfType(context, RetryableException.class);
+        ignoreExceptionOfType(context, ResourceNotFoundException.class);
+
+        IotDataPlaneClientFactory factory = spy(kernel.getContext().get(IotDataPlaneClientFactory.class));
+        kernel.getContext().put(IotDataPlaneClientFactory.class, factory);
+        IotDataPlaneClientWrapper wrapper = spy(new FakeIotDataPlaneClientWrapper(factory));
+        kernel.getContext().put(IotDataPlaneClientWrapper.class, wrapper);
+
+        startNucleusWithConfig(NucleusLaunchUtilsConfig.builder()
+                .configFile("sync.yaml")
+                .mqttConnected(true)
+                .mockCloud(false)
+                .build());
+        BaseSyncStrategy s = kernel.getContext().get(RealTimeSyncStrategy.class);
```

Comment:
Lets use more descriptive names

```suggestion
        BaseSyncStrategy syncStragy = kernel.getContext().get(RealTimeSyncStrategy.class);
```